compiler/ballerina-lang/src/main/java/io/ballerina/projects/JarResolver.java

```diff
@@ -57,17 +62,23 @@ public class JarResolver {
     private final PackageResolution pkgResolution;
     private final PackageContext rootPackageContext;
     private final PrintStream err = System.err;
+    private final List<Diagnostic> diagnostics;
 
     private ClassLoader classLoaderWithAllJars;
 
     JarResolver(JBallerinaBackend jBalBackend, PackageResolution pkgResolution) {
         this.jBalBackend = jBalBackend;
         this.pkgResolution = pkgResolution;
         this.rootPackageContext = pkgResolution.packageContext();
+        this.diagnostics = new ArrayList<>();
+    }
+
+    public List<Diagnostic> diagnostics() {
+        return diagnostics;
     }
 
     // TODO These method names are too long. Refactor them soon
-    public Collection<JarLibrary> getJarFilePathsRequiredForExecution() {
+    public Collection<JarLibrary> getJarFilePathsRequiredForExecution() { // Deprecate
```

Comment:
We should declare a class level variable for `jarFilePathsRequiredForExecution` and return it if it is not null.

Reply:
Added